src/components/textbox/textbox.pw.tsx

```diff
@@ -0,0 +1,1127 @@
+          "min-height"
+        );
+        expect(tokenValues[0]).toBe(token);
+      });
+    }
+  );
+
+  [
+    ["background", "--colorsUtilityYang100"],
+    ["border", "--colorsUtilityMajor300"],
+  ].forEach(([cssProp, token]) => {
+    test(`should render with correct ${cssProp} token`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent />);
+
+      const tokenValues = await getDesignTokensByCssProperty(
+        page,
+        textbox(page),
+        cssProp
+      );
+      expect(tokenValues[0]).toBe(token);
+    });
+  });
+
+  test("should render with isOptional prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent isOptional />);
+
+    const label = getDataElementByValue(page, "label");
+    const labelParent = await label.evaluateHandle((element: Element) => {
+      return element.parentElement;
+    });
+
+    const contentValue = await page.evaluate(
+      (element) =>
+        window.getComputedStyle(element, ":after").getPropertyValue("content"),
+      labelParent
+    );
+    expect(contentValue).toContain("(optional)");
+  });
+
+  test("should render with data-component prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent data-component={CHARACTERS.STANDARD} />);
+
+    await expect(
+      getDataComponentByValue(page, CHARACTERS.STANDARD)
+    ).toBeVisible();
+  });
+
+  test("should render with data-element prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent data-element={CHARACTERS.STANDARD} />);
+
+    await expect(
+      getDataElementByValue(page, CHARACTERS.STANDARD)
+    ).toBeVisible();
+  });
+
+  test("should render with data-role prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent data-role={CHARACTERS.STANDARD} />);
+
+    await expect(getDataRoleByValue(page, CHARACTERS.STANDARD)).toBeVisible();
+  });
+
+  test("should render with id prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent id={CHARACTERS.STANDARD} />);
+
+    await expect(textboxInput(page)).toHaveId(CHARACTERS.STANDARD);
+  });
+
+  testData.forEach((labelVals) => {
+    test(`should render with label prop ${labelVals}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<TextboxComponent label={labelVals} />);
+
+      const label = getDataElementByValue(page, "label");
+
+      await expect(label).toHaveText(labelVals);
+    });
+  });
+
+  test("should render with labelInline prop", async ({ mount, page }) => {
+    await mount(<TextboxComponent labelInline />);
+
+    const label = getDataElementByValue(page, "label");
+    const labelParent = await label.evaluateHandle((element: Element) => {
+      return element.parentElement;
+    });
+
+    const justifyContentValue = await page.evaluate(
+      (element) =>
+        window.getComputedStyle(element).getPropertyValue("justify-content"),
+      labelParent
+    );
+    expect(justifyContentValue).toEqual("flex-end");
```

Comment:
Regarding our conversation yesterday, the following code does the same verification more simply, with the caveat that it uses `locator("..")` to find the parent element:
``const labelParent = getDataElementByValue(page, "label").locathor("..");
await expect(labelParent).toHaveCSS("justify-content", "flex-end");``